packages/vats/src/ibc.js

```diff
@@ -61,6 +62,180 @@ const prepareAckWatcher = zone => {
   return makeAckWatcher;
 };
 
+/**
+ * @param {import('@agoric/base-zone').Zone} zone
+ * @param {ReturnType<import('@agoric/vow').prepareVowTools>} powers
+ * @param {ReturnType<prepareProtocolImplAttemptWatcher>} makeProtocolImplAttemptWatcher
+ */
+const prepareProtocolImplInboundWatcher = (
+  zone,
+  { watch },
+  makeProtocolImplAttemptWatcher,
+) => {
+  const makeProtocolImplInboundWatcher = zone.exoClass(
```

Comment:
This watcher can be a singleton, too.

```suggestion
 * @param {ReturnType<makeProtocolImplAttemptWatcher>} protocolImplAttemptWatcher
 */
const makeProtocolImplInboundWatcher = (
  zone,
  { watch },
  protocolImplAttemptWatcher,
) => {
  const protocolImplInboundWatcher = zone.exo(
```